commons/src/test/java/org/infinispan/commons/dataconversion/MediaTypeTest.java

```diff
@@ -17,105 +19,170 @@
    }
 
    @Test
-   public void testQuotedParams() throws Exception {
+   public void testQuotedParams() {
       MediaType mediaType = MediaType.fromString("application/json; charset=\"UTF-8\"");
 
       assertMediaTypeWithParam(mediaType, "application", "json", "charset", "UTF-8");
    }
 
    @Test
-   public void testQuotedParams2() throws Exception {
+   public void testQuotedParams2() {
       MediaType mediaType = MediaType.fromString("application/json; charset='UTF-8'");
 
       assertMediaTypeWithParam(mediaType, "application", "json", "charset", "UTF-8");
    }
 
    @Test
-   public void testUnQuotedParam() throws Exception {
+   public void testUnQuotedParam() {
       MediaType mediaType = MediaType.fromString("application/json; charset=UTF-8");
 
       assertMediaTypeWithParam(mediaType, "application", "json", "charset", "UTF-8");
    }
 
    @Test
-   public void testToString() throws Exception {
+   public void testToString() {
+      assertEquals("application/xml", new MediaType("application", "xml", createMap(new MapEntry("q", "0.9"))).toString());
       assertEquals("text/csv", new MediaType("text", "csv").toString());
       assertEquals("foo/bar; a=2", new MediaType("foo", "bar", createMap(new MapEntry("a", "2"))).toString());
       assertEquals("foo/bar; a=2; b=1; c=2", new MediaType("foo", "bar",
             createMap(new MapEntry("a", "2"), new MapEntry("b", "1"), new MapEntry("c", "2"))).toString());
-
+      assertEquals("a/b; p=1", MediaType.fromString("a/b; p=1; q=2;").printExcludingParam("q"));
    }
 
    @Test
-   public void testUnQuotedParamWithSpaces() throws Exception {
+   public void testUnQuotedParamWithSpaces() {
       MediaType mediaType = MediaType.fromString("application/json ; charset= UTF-8");
 
       assertMediaTypeWithParam(mediaType, "application", "json", "charset", "UTF-8");
    }
 
    @Test(expected = EncodingException.class)
-   public void testWrongQuoting() throws Exception {
+   public void testWrongQuoting() {
       MediaType.fromString("application/json ; charset= \"UTF-8");
    }
 
    @Test
-   public void testMultipleParameters() throws Exception {
+   public void testMultipleParameters() {
       MediaType mediaType = MediaType.fromString("application/json ; charset=UTF-8; param1=value1; param2 = value2");
       assertMediaTypeWithParams(mediaType, "application", "json",
             new String[]{"charset", "param1", "param2"},
             new String[]{"UTF-8", "value1", "value2"});
    }
 
    @Test(expected = EncodingException.class)
-   public void testMultipleParametersWrongSeparator() throws Exception {
+   public void testMultipleParametersWrongSeparator() {
       MediaType.fromString("application/json ; charset=UTF-8; param1=value1, param2 = value2");
    }
 
+   @Test
+   public void testParseWeight() {
+      MediaType mediaType = MediaType.fromString("application/json ; q=0.8");
+      assertEquals(0.8, mediaType.getWeight(), 0.0);
```

Comment:
```0.8``` is not the message, it's the expected value. This is because ```assertEquals(double, double)``` is deprecated in junit in favour of a 3 param method with a ```delta```

Reply:
Ah gotcha.